### PR TITLE
Add rcParams to enable/disable minor ticks on axes separately issue #3024

### PR DESCRIPTION
**Pull request by crazym**
Hi,

I added two rcParams "xtick.minor.visible" and "ytick.minor.visible" to enable/disable minor ticks for x/y axis respectively, as mentioned in the issue #3024.

EDIT:
After reading a few PRs I checked the coding guide, updated the tests for these two new rcParams and squashed everything into one commit. Hope it helps. :)

Thanks
Minty

### CODE REVIEW

**Comment by tacaswell** [comment]
I don't think we need svg or pdf tests on this one, the png should be enough.

**Comment by tacaswell** [comment]
Can you also add a note to https://github.com/matplotlib/matplotlib/tree/master/doc/users/whats_new ?

**Comment by crazym** [comment]
Updated, let me know if there are any other problems :)

File: doc/users/whats_new/rcparams.rst [hunk 1]
```diff
@@ -1,3 +1,9 @@
+Added ``xtick.minor.visible`` and ``ytick.minor.visible`` key to rcParams
+`````````````
```
Comment (tacaswell) [inline doc/users/whats_new/rcparams.rst]: The tick line should match the length of the title line

File: doc/users/whats_new/rcparams.rst [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+Two new keys to control the minor ticks on x/y axis respectively, default set to False(hide minor ticks on the axis).
```
Comment (tacaswell) [inline doc/users/whats_new/rcparams.rst]: It is less 'hide' and more 'never shown' as by the current default is a `NullLocator`

File: doc/users/whats_new/rcparams.rst [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+
+When enabled, it will present the automatic minor ticks.
```
Comment (tacaswell) [inline doc/users/whats_new/rcparams.rst]: "When ``True`` the minor ticks are shown and located via a``mticker.AutoMinorLocator()``."

**Comment by crazym** [comment]
Updated, thanks for the suggestion!

**Comment by tacaswell** [comment]
Great!  I will merge this once we hear back from travis.

**Comment by radlinsky** [comment]
Thanks crazym for adding those rcParams!

Is it possible to update this help documentation?
http://matplotlib.org/users/customizing.html

There are many rcParams that aren't described in the help docs including the two described in this thread ...

(sorry if it this is an inappropriate place to post this)

**Comment by tacaswell** [comment]
@crazym PRs to the docs are always welcome!

**Comment by tacaswell** [comment]
@crazym, sorry, I meant to ping @radlinsky in my last message

**Comment by radlinsky** [comment]
@tacaswell sorry for the newbie question, but what does PR mean, and if you're saying I can contribute to the help docs, how would I go about doing do? Thanks!

**Pull request merged**